### DISCUSSION TITLE
Welcome to c4sharp Discussions!

### DISCUSSION THREAD

**Discussion by yanjustino** [8T4/c4sharp#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Gullinvargr** [comment]
The C4 Model suggests that if you're going to use colors, shapes (other than plain boxes) different lines for different kinds of relationships, etc., that you should have a legend to explain these rather than expecting people looking at the diagram to guess. Is there any way to create such a legend with c4sharp? I tried creating a boundary box with generic objects of the various types, but it tends to be a bit big and you can't add relationship lines. Any suggestions?

**Reply by yanjustino** [reply]
Hi, @Gullinvargr! To expose the legend you should use the property ShowLegend in your code, like this:

```c#
public class ComponentDiagramSample : ComponentDiagram
{
    protected override string Title => "Internet Banking System API Application";
    protected override DiagramLayout FlowVisualization => DiagramLayout.LeftRight;
    
    //To show the legend
    protected override bool ShowLegend => true;
...
```

**Reply by Gullinvargr** [reply]
Thanks!

**Comment by Gullinvargr** [comment]
Another question. Is there any way to apply the standard database shape (cylinder)? PlantUML has that shape but I can't figure out a way to get to it with c4sharp.

**Reply by yanjustino** [reply]
Yep!!! To shape a container as a database shape you should use Database class, like this:
see our sample code [here](https://github.com/8T4/c4sharp/blob/main/samples/ModelDiagrams/Diagrams/ComponentDiagramSample.cs)

``` c#

    public static Database SqlDatabase => new (
        Alias: "Database",
        Label: "SqlDatabase",
        Description: "Stores user registration information, hashed auth credentials, access logs, etc.",
        Technology: "SQL Database"
    );
```

**Comment by Gullinvargr** [comment]
Back again. 😉 Can text in a container be aligned left instead of centered? For longer descriptions I think it would be more readable. It doesn't seem to be an option in themes.
